### PR TITLE
Add search transformer [Preview] [PD-614]

### PR DESCRIPTION
**Pull request by darrint**
This is working search query transformer which passes known tests.

Docs are inline in the code.

~~It isn't hooked up to anything yet.~~

~~Edit: It is naively hooked up to microsites now.~~

It is ready for review.

### CODE REVIEW

File: seo/search_transformer.py [hunk 1]
```diff
@@ -0,0 +1,585 @@
+import re
+from sets import Set
```
Comment (koonsies) [inline seo/search_transformer.py]: My IDE is informing me that Set is deprecated as of v2.6, replaced by native "set".

Comment (darrint) [inline seo/search_transformer.py]: Fixed.

**Comment by koonsies** [comment]
TLDR: Take input from the "What" search field on MyJobs SEO sites and convert into Solr/Lucene readable format.

File: seo/search_transformer.py [hunk 2]
```diff
@@ -0,0 +1,584 @@
...
+term_re = re.compile(r'(%s)(.*)' % raw_term_pattern, re.U)
+
+token_peekable = peekable(Token('eof', ''))
+
+
+@token_peekable
+def tokenize(input_query):
+    current = input_query
+    while(current != ''):
+        # eat ws
```
Comment (koonsies) [inline seo/search_transformer.py]: Good method of killing whitespace, but may as well extend the comment to "eat whitespace". I personally like more verbose variable/function names, but in lieu of that, verbose comments are important.

File: seo/search_transformer.py [hunk 3]
```diff
@@ -0,0 +1,584 @@
...
+        if match:
+            current = match.group(2)
+            yield Token('term', match.group(1), ['quote'])
+            continue
+
+        # try to match prefix - special behavior
+        match = dash_re.match(current)
+        if match:
+            current = match.group(2)
+            yield Token('pdash', match.group(1).lower())
```
Comment (koonsies) [inline seo/search_transformer.py]: This calls lower() on a regex group that could only be a "-"

File: seo/search_transformer.py [hunk 4]
```diff
@@ -0,0 +1,584 @@
...
+    def make_peekable(fn):
+        @wraps(fn)
+        def peekable_fn(*args):
+            return Peekable(fn(*args), last)
+        return peekable_fn
+    return make_peekable
+
+# Tokenizer -------------------------------
+
+and_ops = set(["and", "&", "-"])
```
Comment (aspidites) [inline seo/search_transformer.py]: Any reasn to not just use set literals here?

File: seo/__init__.py [hunk 5]
```diff
@@ -1 +0,0 @@
-import signals
```
Comment (JLMcLaugh) [inline seo/__init__.py]: Removing this line breaks signals tests for me, but it looks like they're passing on Jenkins. What am I missing?

**Pull request merged**